File: lehuan-sellergoods-interface/src/main/java/com/lehuan/sellergoods/service/BrandService.java
package com.lehuan.sellergoods.service;

import com.lehuan.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

/**
 * 商品品牌接口
 * @author Administrator
 *
 */
public interface BrandService {
//	返回品牌列表
	public List<TbBrand> findAll();

	/**
	 * 返回品牌分页列表
	 * @param pageNum   当前页面
	 * @param pageSize	每页记录数
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	/**
	 * 按条件返回品牌分页列表,因为有多个查询条件，所以用TbBrand类型参数接收
	 * @param pageNum   当前页面
	 * @param pageSize	每页记录数
	 * @return
	 */
	public PageResult findPage(TbBrand brand, int pageNum, int pageSize);
//	增加品牌接口
	public void add(TbBrand brand);
//	根据ID查询实体
	public TbBrand findOne(Long id);
//	修改
	public void update(TbBrand brand);
//	批量删除
	public void delete(Long[] ids);
	/**
	 * 品牌下拉框数据
	 */
	List<Map> selectOptionList();
}

File: lehuan-manager-web/src/main/webapp/js/service/brandService.js
//在品牌服务层定义好一些方法,写好后在控制层中调用,也可以被其他控制层调用
app.service("brandService", function ($http) {
    this.findAll = function () {
        return $http.get('../brand/findAll.do');
    }
    this.findPage = function (page, rows) {
        return $http.get('../brand/findPage.do?page=' + page + '&size=' + rows);
    }
    this.findOne = function (id) {
        return $http.get('../brand/findOne.do?id=' + id);
    }
    this.add = function (entity) {
        return $http.post('../brand/add.do', entity);
    }
    this.update = function (entity) {
        return $http.post('../brand/update.do', entity);
    }
    this.dele = function (ids) {
        return $http.get('../brand/delete.do?ids=' + ids);
    }
    this.search = function (page, rows, searchEntity) {
        return $http.post('../brand/search.do?page=' + page + '&size=' + rows, searchEntity);
    }
    //下拉列表数据
    this.selectOptionList=function(){
        return $http.get('../brand/selectOptionList.do');
    }
});